pkg/finality-grandpa/bridge_state.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"sync"
+)
+
+type waker struct {
+	mtx    sync.RWMutex
+	wakeCh chan any
+}
+
+func newWaker() *waker {
+	return &waker{wakeCh: make(chan any, 1000)}
+}
+
+func (w *waker) wake() {
+	w.mtx.RLock()
+	defer w.mtx.RUnlock()
+	if w.wakeCh == nil {
+		return
+	}
+	go func() {
+		select {
+		case w.wakeCh <- nil:
+		default:
+		}
+	}()
+}
+
+func (w *waker) channel() chan any {
+	return w.wakeCh
+}
+
+func (w *waker) register(waker *waker) {
+	w.mtx.Lock()
+	defer w.mtx.Unlock()
+	w.wakeCh = waker.wakeCh
+}
+
+// round state bridged across rounds.
+type bridged[Hash, Number any] struct {
+	inner RoundState[Hash, Number]
+	// registered map[chan State[Hash, Number]]any
+	waker *waker
+	sync.RWMutex
+}
+
+func (b *bridged[H, N]) update(new RoundState[H, N]) {
+	b.Lock()
+	b.inner = new
+	b.waker.wake()
+	b.Unlock()
+}
+
+func (b *bridged[H, N]) get(waker *waker) RoundState[H, N] {
+	b.RLock()
+	defer b.RUnlock()
+	b.waker.register(waker)
+	return b.inner
+}
```

Comment:
could you explain why `get` updates the current waker of the bridge?

Reply:
`bridged` is used as the `bridged` attribute for both `priorView` and `latterView`.  If `priorView.update` gets called essentially updating the `RoundState` of `priorView`, it will wake the function that started this whole thing which is `Voter.poll` to essentially update.  The `wakerCh` is essentially emulating rust futures.  If a previous round is updated, then it triggers `Voter.poll` to run again.